pipeline/beam_tables.py

```diff
@@ -448,15 +386,151 @@ def _flatten_measurement(
 
       if 'Received' in result:
         received = result.get('Received', '')
-        received_fields = _parse_received_data(
-            received, scan['Blocked'], blockpage_matcher=blockpage_matcher)
+        received_fields = self._parse_received_data(received, scan['Blocked'])
         row.update(received_fields)
 
       if 'Error' in result:
         row['error'] = result['Error']
 
       yield row
 
+  def _process_satellite(self, filename: str, scan: Any,
+                         random_measurement_id: str) -> Iterator[Row]:
+    """Process a line of Satellite data.
+
+    Args:
+      filename: a filepath string
+      scan: a loaded json object containing the parsed content of the line
+      random_measurement_id: a hex id identifying this individual measurement
+
+    Yields:
+      Rows
+    """
+    date = re.findall(r'\d\d\d\d-\d\d-\d\d', filename)[0]
+    if date < "2021-03":
+      yield from self._process_satellite_v1(date, scan, random_measurement_id)
+    else:
+      yield from self._process_satellite_v2(scan, random_measurement_id)
+
+  def _process_satellite_v1(  # pylint: disable=no-self-use
+      self, date: str, scan: Any, random_measurement_id: str) -> Iterator[Row]:
+    """Process a line of Satellite data.
+
+    Args:
+      date: a date string YYYY-mm-DD
+      scan: a loaded json object containing the parsed content of the line
+      random_measurement_id: a hex id identifying this individual measurement
+
+    Yields:
+      Rows
+    """
+    row = {
+        'domain': scan['query'],
+        'ip': scan['resolver'],
+        'date': date,
+        'error': scan.get('error', None),
+        'blocked': not scan['passed'] if 'passed' in scan else None,
+        'success': 'error' not in scan,
+        'received': None,
+        'measurement_id': random_measurement_id,
+    }
+    received_ips = scan.get('answers')
+    yield from self._process_received_ips(row, received_ips)
+
+  def _process_satellite_v2(self, scan: Any,
```

Comment:
@avirkud For Satellite-v2, we might want to do blockpage matching using the `blockpages.json` file.

Reply:
Yes, I can add this once the current Satellite pipeline is working in the cloud.